cmd/katapult/client_test.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/krystal/go-katapult"
+	"github.com/krystal/katapult-cli/config"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_newClient(t *testing.T) {
+	tests := []struct {
+		name string
+
+		apiKey string
+		apiURL string
+		err    string
+	}{
+		{
+			name:   "blank API URL",
+			apiKey: "test",
+		},
+		{
+			name:   "invalid URL",
+			err:    "invalid API URL: @@:",
+			apiURL: "@@:",
+		},
+		{
+			name:   "both values",
+			apiKey: "test",
+			apiURL: "https://example.com",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c, err := newClient(&config.Config{
+				APIKey: tt.apiKey,
+				APIURL: tt.apiURL,
+			})
+			switch {
+			case err == nil:
+				// Ignore this. Jump out here.
+			case tt.err != "":
+				assert.Equal(t, tt.err, err.Error())
+				return
+			default:
+				t.Fatal(err)
+			}
```

Comment:
The `err == nil` case here allows this test to pass for a test case that expects an error to happen if `err` is `nil`.

Hence I suggest we make the assertion dependent purely based on the `tt.err` value, and it means we can simplify the switch statement to a single if statement:

```suggestion
			if tt.err != "" {
				require.EqualError(t, err, tt.err)
				return
			}

			require.NoError(t, err)
```